crates/turbo-tasks/src/generics/index_map.rs

```diff
@@ -0,0 +1,71 @@
+use anyhow::Result;
+use indexmap::IndexMap;
+// This specific macro identifier is detected by turbo-tasks-build.
+use turbo_tasks_macros::generic_type as __turbo_tasks_internal_generic_type;
+
+use crate::{
+    self as turbo_tasks,
+    debug::{ValueDebug, ValueDebugFormat, ValueDebugString},
+    ValueDefault, Vc,
+};
+
+__turbo_tasks_internal_generic_type!(<K, V>, IndexMap<Vc<K>, Vc<V>>);
+
+#[turbo_tasks::function]
+async fn index_map_len(index_map: Vc<IndexMap<Vc<()>, Vc<()>>>) -> Result<Vc<usize>> {
+    let index_map = index_map.await?;
+    Ok(Vc::cell(index_map.len()))
+}
```

Comment:
I've added these methods because I recalled a few places where we did something similar on a `#[turbo_tasks::value(transparent)]` vector of `Vc`s, but I actually can't find a good example. So perhaps we don't need these indeed and should recommend the `.await?.method()` form instead.